File: common/src/main/java/com/example/common/validator/LongValidator.java
package com.example.common.validator;

public class LongValidator implements Validator {
    private final long value;
    private boolean result;

    public LongValidator(long value) {
        this.result = true;
        this.value = value;
    }

    public LongValidator greaterThanZero() {
        this.result &= this.value > 0;
        return this;
    }

    @Override
    public boolean isValid() {
        return this.result;
    }
}

File: services/common/src/main/kotlin/com/example/common/validator/LongValidator.kt
package com.example.common.validator

class LongValidator(private val value: Long): Validator {

    private var result = true

    fun greaterThanZero(): LongValidator {
        result = result and (value > 0)
        return this
    }

    override fun isValid(): Boolean {
        return result
    }
}